### DISCUSSION TITLE
Disable Expect100Continue and how to sub-class S3 Requests

### DISCUSSION THREAD

**Discussion by DetlefGolze** [aws/aws-sdk-net#3356] [Q&A] [answered]
I want to disable the Expect100Continue header and thought that it is as easy as creating a derived class of PutObjectRequest which already has an overridable property for this:

```
	public partial class MyPutObjectRequest : PutObjectRequest 
	{
		public MyPutObjectRequest()
		{
			// This is the default setting in base library
			//
			MyExpect100Continue = true;
		}

		public bool MyExpect100Continue { get; set; }

                protected override bool Expect100Continue
                {  get  { return MyExpect100Continue; }    }
	}
```

So far so good. This compiles and appears to do what I expect. However, the AmazonS3ControlEndpointResolver uses the name of the request class to implement some request specific behavior causing requests to fail which use an unknown type.

So, two questions:
Is there a better way to disable Expect100Continue? I suppose it is not used as intended anyway and only generates overhead.

Is there an official way to create sub-classes of Request classes? Except of the above example I was hoping that I can use this to pass data from and to an AfterResponseEvent. My S3Client is shared by many threads and this would make life much easier.
We could add a request property which returns the expected RequestName or just use something like this in AmazonS3ControlEndpointResolver:
```
    if (request is PutObjectRequest)
    {
    }
```

**Comment by ashishdhingra** [comment, marked as answer]
@DetlefGolze We have converted this discussion into a new issue https://github.com/aws/aws-sdk-net/issues/3375 which would be investigated by team.
